prov/net/src/xnet_progress.c

```diff
@@ -765,11 +801,35 @@ void xnet_progress_async(struct xnet_ep *ep)
 static void xnet_progress_cqe(struct xnet_uring *uring,
 			      ofi_io_uring_cqe_t *cqe)
 {
+	struct xnet_xfer_entry *tx_entry;
 	struct ofi_bsock *bsock;
+	struct xnet_ep *ep;
 
 	assert(xnet_io_uring);
 	bsock = (struct ofi_bsock *) cqe->user_data;
 	assert(bsock);
+
+	if (&uring->ring == bsock->sockapi->tx_io_uring) {
```

Comment:
Is this check to distinguish sends from receives (in the future)?

Reply:
Right, this check is used to distinguish sends from receives. The `else` block for receives will be implemented in another PR.